src/datatype/index.js

```diff
@@ -239,12 +239,17 @@ export class DataType extends TypedEmitter {
     return this.getByDocId(docId)
   }
 
-  /**
-   * @param {Parameters<import('drizzle-orm/better-sqlite3').BetterSQLite3Database['select']>[0]} fields
-   */
-  async [kSelect](fields) {
+  async [kSelect]() {
     await this.#dataStore.indexer.idle()
-    return this.#db.select(fields).from(this.#table)
+    const result = this.#db.select().from(this.#table)
+    // [The result of `from()` is awaitable.][0] We don't want this because we
+    // want to be able to await the result of this method and then call
+    // `.where()` on the result.
+    //
+    // As a workaround, we remove `.then` from the result.
+    //
+    // [0]: https://github.com/drizzle-team/drizzle-orm/commit/c063144dc08726cc15323582fe377210329e579e
+    return removeThen(result)
```

Comment:
I would love a better way of doing this...

Reply:
spent a brief moment looking into this and there isn't really a great way from what I can tell (unless we want to change the way we write queries 😬)

do other promise-related fields need to be removed as well?

Reply:
Good idea. Done in 5c249cd208e0ace06cf72f7638c28e6fc334e953.